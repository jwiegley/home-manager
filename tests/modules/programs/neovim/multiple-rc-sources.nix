{ config, pkgs, ... }:

{
  config = {
    programs.neovim = {
      enable = true;
      extraConfig = ''
        set sw=4
      '';
      extraLuaConfig = ''
        vim.opt.expandtab = false
      '';
      plugins = [{
        plugin = pkgs.emptyDirectory;
        type = "lua";
        config = ''
          vim.g.test_plugin2 = 1
        '';
      }];
    };

    nmt.script = ''
      initLua="$TESTED/home-files/.config/nvim/init.lua"
      initLuaNormalized="$(normalizeStorePaths "$initLua")"
      assertFileExists "$initLua"
      assertFileContent "$initLuaNormalized" "${./multiple-rc-sources.expected}"
    '';
  };
}

