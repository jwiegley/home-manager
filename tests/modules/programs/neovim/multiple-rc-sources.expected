vim.cmd [[source /nix/store/00000000000000000000000000000000-nvim-init-home-manager.vim]]

vim.opt.expandtab = false

vim.g.test_plugin2 = 1
